src/components/Settings/RouterPreferenceSettings/index.tsx

```diff
@@ -49,18 +43,13 @@ export default function RouterPreferenceSettings() {
         // If UniswapX-by-default is enabled we need to render this as active even if routerPreference === RouterPreference.API
         // because we're going to default to the UniswapX quote.
         // If the user manually toggles it off, this doesn't apply.
-        isActive={uniswapXInEffect}
+        isActive={routerPreference === RouterPreference.X}
         toggle={() => {
-          if (uniswapXInEffect) {
-            if (isUniswapXDefaultEnabled) {
-              // We need to remember if a opts out of UniswapX, so we don't request UniswapX quotes.
-              dispatch(updateOptedOutOfUniswapX({ optedOutOfUniswapX: true }))
-            } else {
-              // We need to remember if a user disables Uniswap X, so we don't show the opt-in flow again.
-              dispatch(updateDisabledUniswapX({ disabledUniswapX: true }))
-            }
+          if (routerPreference === RouterPreference.X) {
+            // We need to remember if a opts out of UniswapX, so we don't request UniswapX quotes.
```

Comment:
i think we can get rid of `optedOutOfUniswapX` user state, wdyt?

Reply:
and then just simplify this whole thing to just line 52

Reply:
actually yeah we can get rid of it, it's only used in the migration.